### PR TITLE
GL-705: Remove add code from permutation key

### PR DESCRIPTION
**Pull request by rasikasri**
### Jira link

[GL-705](https://transformuk.atlassian.net/browse/GL-705)

### What?

I have added/removed/altered:

- [ ] Removed additional code key from the permutation key

### Why?

I am doing this because:

- We don't need to create separate CA for each additional codes. Instead, we need to group them into a one CA

### CODE REVIEW

File: app/presenters/api/v2/green_lanes/category_assessment_presenter.rb [hunk 1]
```diff
@@ -77,7 +76,8 @@ def certificates
         end
 
         def additional_codes
-          Array.wrap(additional_code)
+          additional_codes = measures.map(&:additional_code).compact
```
Comment (willfish) [inline app/presenters/api/v2/green_lanes/category_assessment_presenter.rb:79]: I'm guessing the answer is yes, but has this association for measures been eager loaded upstream of this call?

**Comment by willfish** [comment]
My basic understanding of additional code is they basically decide whether the current measure is applicable (in most cases) or whether a certain duty rate applies.

So what we're saying here is that they're non-deterministic in terms of whether the current category  assessment abstraction applies for a given commodity.

So I guess this means that we're basically assuming that whether the additional code is relevant comes only after we ask the user for more information in the frontend UI and excludes specific measures, here?

File: app/presenters/api/v2/green_lanes/category_assessment_presenter.rb (hunk 1 above)
Comment (rasikasri) [reply app/presenters/api/v2/green_lanes/category_assessment_presenter.rb:79]: Yes, here you can see additional_code is in the eager loading plan
https://github.com/trade-tariff/trade-tariff-backend/blob/GL-705_remove_add_code_from_permutation_key/app/services/green_lanes/fetch_goods_nomenclature_service.rb

**Comment by rasikasri** [comment]
> My basic understanding of additional codes is they basically decide whether the current measure is applicable (in most cases) or whether a certain duty rate applies.
> 
> So what we're saying here is that they're non-deterministic in terms of whether the current category assessment abstraction applies for a given commodity essentially because we need more information.
> 
> So I guess this means that we're basically assuming that whether the additional code is relevant comes only after we ask the user for more information in the frontend UI and excludes specific measures, here?

We have multiple measures for same measure type with different additional_code attached to the measure record. So if we have additional_code in the permutation key, we gonna end up with number of category assessments for the same category per each additional code. Instead we want to have one CA aggigating all the additional_code into it. Then user can choose if there is a additional_code that applicable to the user and then exempt the category.

**Comment by willfish** [comment]
Got it. Okay

**Pull request merged**